src/main/java/otter/jet/reader/ReaderService.java

```diff
@@ -93,21 +97,35 @@ private void continuouslyReadMessages(
             Message message = subscription.nextMessage(100);
             // Print the message
             if (message != null) {
-                try {
-                    DeserializedMessage deserializedMessage =
-                            messageDeserializer.deserializeMessage(ByteBuffer.wrap(message.getData()));
-                    ReadMessage msg =
-                            new ReadMessage(
-                                    message.getSubject(),
-                                    deserializedMessage.name(),
-                                    deserializedMessage.content(),
-                                    message.metaData().timestamp().toLocalDateTime());
-                    messageStore.add(msg);
+                LocalDateTime messageTimestamp = message
+                        .metaData()
+                        .timestamp()
+                        .toLocalDateTime();
+
+                if (messageTimestamp.isAfter(beginTimestamp)) {
+                    deserializeMessage(messageDeserializer, message, messageTimestamp);
+                } else {
+                    LOG.warn("Timestamp from message {}, smaller then begin timestamp {}, message {} will not be process", messageTimestamp, beginTimestamp, message.getSID());
                     message.ack();
-                } catch (Exception e) {
-                    LOG.warn("Unable to deserialize message", e);
                 }
             }
         }
     }
+
+    private void deserializeMessage(MessageDeserializer messageDeserializer, Message message, LocalDateTime timestamp) {
```

Comment:
how about using `MessageDeserializer` filed from object directly instead of passing it as param ?